kernelci/config/build.py

```diff
@@ -49,6 +49,33 @@ def url(self):
         return self._url
 
 
+class Reference(YAMLObject):
```

Comment:
I was initially expecting a plain dictionary but now I think it makes sense to have a class for this.